src/Trace/TraceClient.php

```diff
@@ -0,0 +1,209 @@
+ * your applications and display it in the Google Cloud Platform Console.
+ * Find more information at [Stackdriver Trace API docs](https://cloud.google.com/trace/docs/).
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ *
+ * $trace = $cloud->trace();
+ * ```
+ *
+ * ```
+ * // TraceClient can be instantiated directly
+ * use Google\Cloud\Trace\TraceClient;
+ *
+ * $trace = new TraceClient();
+ * ```
+ */
+class TraceClient
+{
+    use ClientTrait;
+
+    const FULL_CONTROL_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
+    const READ_ONLY_SCOPE = 'https://www.googleapis.com/auth/trace.readonly';
+
+    /**
+     * @var ConnectionInterface $connection Represents a connection to Trace
+     */
+    protected $connection;
+
+    /**
+     * Create a Trace client.
+     *
+     * @param array $config [optional] {
+     *     Configuration options.
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type CacheItemPoolInterface $authCache A cache used storing access
+     *           tokens. **Defaults to** a simple in memory implementation.
+     *     @type array $authCacheOptions Cache configuration options.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *           Only valid for requests sent over REST.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Sends a Trace log in a simple fashion.
+     *
+     * @param  Trace $trace The trace log to send.
+     * @return Trace
+     */
+    public function insertTrace(Trace $trace)
+    {
+        $traces = $this->patchTraces([$trace]);
+        return $traces[0];
+    }
+
+    /**
+     * Sends multiple Trace logs in a simple fashion.
+     *
+     * @param Trace[] $traces The trace logs to send.
+     * @return Trace[] Array of new or updated traces.
+     */
+    public function patchTraces(array $traces)
```

Comment:
Throughout the rest of the library, we have `<action>`, `<action>Batch` names. In this case it should be `insertTrace`, `insertTraceBatch`.